components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -127,11 +132,62 @@ protected String getAuthenticateUser(OAuthClientResponse token) {
     }
 
     /**
+     *
+     * @param token
+     * @return
+     */
+    protected String getUserClaimUri(OAuthClientResponse token) {
+        return null;
+    }
+
+    /**
      * @param token
      * @return
      */
     protected Map<ClaimMapping, String> getSubjectAttributes(OAuthClientResponse token) {
-        return new HashMap<ClaimMapping, String>();
+
+        Map<ClaimMapping, String> claims = new HashMap<ClaimMapping, String>();
+
+        try {
+
+            String accessToken = token.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            String url = getUserClaimUri(token);
```

Comment:
This logic is wrong. getUserClaimUri only should give the claim URI you need to send as the username to SP. The UserInfo URL should come from a separate method. For that we must add a new protected method in OIDCAuthenticator and override it in specific authenticators such as Google and Yahoo.